### PR TITLE
HAL level profiling with counters for PCIe and Zynq

### PR DESCRIPTION
**Pull request by IshitaGhosh**

### CODE REVIEW

File: src/runtime_src/core/pcie/linux/shim.cpp [hunk 1]
```diff
@@ -1429,6 +1432,14 @@ int shim::xclReadTraceData(void* traceBuf, uint32_t traceBufSz, uint32_t numSamp
     return size;
 }
 
+
+#if 0
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/linux/shim.cpp]: if this is really needed for future use, please create a proper macro name - #ifdef XXX. Put #if 0 in multiple files for related code is not that clean.

File: src/runtime_src/xrt/device/halops2.h [hunk 2]
```diff
@@ -118,6 +118,9 @@ class operations
                                       bool shared);
   typedef int (* closeContextFuncType)(xclDeviceHandle handle, const uuid_t xclbinId, unsigned ipIndex);
 
+  // xdp hal plugin switch
+//  typedef int (* switchProfilingFuncType)(xclDeviceHandle handle);
```
Comment (maxzhen) [inline src/runtime_src/xrt/device/halops2.h]: remove?

File: src/runtime_src/core/pcie/linux/shim.cpp (hunk 1 above)
Comment (IshitaGhosh) [reply src/runtime_src/core/pcie/linux/shim.cpp]: Done

File: src/runtime_src/xrt/device/halops2.h (hunk 2 above)
Comment (IshitaGhosh) [reply src/runtime_src/xrt/device/halops2.h]: Done

**Pull request merged**